src/transformers/trainer.py

```diff
@@ -2737,44 +2737,31 @@ def save_model(self, output_dir: Optional[str] = None, _internal_call: bool = Fa
             or self.fsdp is not None
             or self.is_fsdp_enabled
         ):
+            state_dict = self.model.state_dict()
+            if self.args.should_save:
+                self._save(output_dir, state_dict=state_dict)
             if self.is_fsdp_enabled:
-                os.makedirs(output_dir, exist_ok=True)
+                if self.args.should_save:
+                    for filename in [WEIGHTS_NAME, SAFE_WEIGHTS_NAME]:
+                        file = os.path.join(output_dir, filename)
+                        if os.path.isfile(file):
+                            os.remove(file)
```

Comment:
Why is this necessary? They will be replaced anyway.

Reply:
Makes sense, resolved